tests/support/client.rs

```diff
@@ -245,40 +271,135 @@ impl hyper::client::connect::Connect for Conn {
     type Error = ::std::io::Error;
     fn connect(&self, _: hyper::client::connect::Destination) -> Self::Future {
         let connected = hyper::client::connect::Connected::new().proxy(self.absolute_uris);
-        Box::new(self.connect_().map(|t| (t, connected)))
+        Box::new(self.initialize_connection().map(|t| (t, connected)))
     }
 }
 
-/// A wrapper around a TcpStream, allowing us to signal when the connection
-/// is dropped.
-struct RunningIo {
-    inner: TcpStream,
-    /// When this drops, the related Receiver is notified that the connection
-    /// is closed.
-    running: oneshot::Sender<()>,
+pub enum ConnectorFuture<F: Future> {
+    Init {
+        future: F,
+        tls: Option<(Arc<ClientConfig>, DNSName)>,
```

Comment:
Again, would prefer to see the type alias used consistently if we're going to continue using one:
```suggestion
        tls: Option<TlsConfig>,
```